internal/signerverifier/git/git.go

```diff
@@ -0,0 +1,96 @@
+// Copyright The gittuf Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package git
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/gittuf/gittuf/internal/gitinterface"
+	"github.com/gittuf/gittuf/internal/signerverifier/sigstore"
+	sigstoresigneropts "github.com/gittuf/gittuf/internal/signerverifier/sigstore/options/signer"
+	"github.com/gittuf/gittuf/internal/signerverifier/ssh"
+	sslibdsse "github.com/gittuf/gittuf/internal/third_party/go-securesystemslib/dsse"
+)
+
+var (
+	ErrSigningMethodNotSpecified = errors.New("signing method not specified in Git configuration")
+	ErrSigningKeyNotSpecified    = errors.New("signing key not specified in Git configuration")
+	ErrUnsupportedX509Method     = errors.New("unsupported X509 certificate specified in Git configuration")
+)
+
+type SigningMethod int
+
+const (
+	SigningMethodGPG SigningMethod = iota
+	SigningMethodSSH
+	SigningMethodX509
+)
+
+// LoadSignerFromGitConfig loads a metadata signer from the signing key
+// specified in the Git config. Currently, only SSH keys and Sigstore are
+// supported.
+func LoadSignerFromGitConfig(repo *gitinterface.Repository) (sslibdsse.SignerVerifier, error) {
+	config, err := repo.GetGitConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	// Attempt to determine what type of key is specified by the user's Git
+	// config
+	var keyType SigningMethod
+	switch config["gpg.format"] {
+	case "gpg":
+		keyType = SigningMethodGPG // nolint:ineffassign
+		return nil, fmt.Errorf("support for GPG keys is not implemented")
+	case "ssh":
+		keyType = SigningMethodSSH
+	case "x509":
+		keyType = SigningMethodX509
+	default:
+		return nil, ErrSigningMethodNotSpecified
+	}
+
+	// Get the path to the signing key
+	signingKey := config["user.signingkey"]
+	if signingKey == "" {
+		return nil, ErrSigningKeyNotSpecified
```

Comment:
isn't this going to be unspecified for gitsign? and gpg for that matter (use default key iirc)?